main.py

```diff
@@ -193,14 +193,25 @@ def upload_part(byte_pos, vault_name, upload_id, part_size, fileobj, file_size,
     print('Uploading part {0} of {1}... ({2:.2%})'.format(
         part_num + 1, num_parts, percentage))
 
-    response = glacier.upload_multipart_part(
-        vaultName=vault_name, uploadId=upload_id, range=range_header,
-        body=part)
+    MAX_ATTEMPTS = 10
+    uploaded = False
+    for i in range(MAX_ATTEMPTS):
+        try:
+            response = glacier.upload_multipart_part(
+                vaultName=vault_name, uploadId=upload_id, range=range_header,
+                body=part)
+            checksum = calculate_tree_hash(part, part_size)
+            assert checksum == response['checksum'], 'Checksums do not match.'
```

Comment:
instead of asserting, just use an `if` condition to check: if checksums match, then break from loop.

Reply:
Yeah, I plan to change some code styles myself anyways.